ivy/functional/ivy/data_type.py

```diff
@@ -538,9 +538,13 @@ def can_cast(
         b: true
     }
     """
-    if isinstance(from_, ivy.Dtype):
-        return (from_, to) in ivy.promotion_table
-    return (from_.dtype, to) in ivy.promotion_table
+    if isinstance(from_, (ivy.Array, ivy.NativeArray)):
```

Comment:
The older approach seemed a bit cleaner actually, unless there's a reason except for the reverse promotions, we could just revert this changes if that's okay?